game/scripts/vscripts/modifiers/modifier_duel_invulnerability.lua

```diff
@@ -0,0 +1,20 @@
+modifier_duel_invulnerability = class({})
+
+function modifier_duel_invulnerability:CheckState()
+	return {
+	[MODIFIER_STATE_INVULNERABLE] = true,
+	[MODIFIER_STATE_MAGIC_IMMUNE] = true,
```

Comment:
I think magic immunity is redundant when they're already invulnerable, but I suppose it doesn't matter anyway.